ldms/src/core/ldms.c

```diff
@@ -2875,6 +2875,276 @@ double ldms_metric_array_get_double(ldms_set_t s, int mid, int idx)
 	return 0;
 }
 
+void ldms_mval_set_char(ldms_mval_t mv, char v)
+{
+	mv->v_char = v;
+}
+
+void ldms_mval_set_u8(ldms_mval_t mv, uint8_t v)
+{
+	mv->v_u8 = v;
+}
+
+void ldms_mval_set_s8(ldms_mval_t mv, int8_t v)
+{
+	mv->v_s8 = v;
+}
+
+void ldms_mval_set_u16(ldms_mval_t mv, uint16_t v)
+{
+	mv->v_u16 = __cpu_to_le16(v);
+}
+
+void ldms_mval_set_s16(ldms_mval_t mv, int16_t v)
+{
+	mv->v_s16 = __cpu_to_le16(v);
+}
+
+void ldms_mval_set_u32(ldms_mval_t mv, uint32_t v)
+{
+	mv->v_u32 = __cpu_to_le32(v);
+}
+
+void ldms_mval_set_s32(ldms_mval_t mv, int32_t v)
+{
+	mv->v_s32 = __cpu_to_le32(v);
+}
+
+void ldms_mval_set_u64(ldms_mval_t mv, uint64_t v)
+{
+	mv->v_u64 = __cpu_to_le64(v);
+}
+
+void ldms_mval_set_s64(ldms_mval_t mv, int64_t v)
+{
+	mv->v_s64 = __cpu_to_le64(v);
+}
+
+void ldms_mval_set_float(ldms_mval_t mv, float v)
+{
+#if LDMS_SETH_F_LCLBYTEORDER == LDMS_SETH_F_LE
+	mv->v_f = v;
+#else
+	*(uint32_t *)&mv->v_f = __cpu_to_le32(*(uint32_t *)&v);
+#endif
+}
+
+void ldms_mval_set_double(ldms_mval_t mv, double v)
+{
+#if LDMS_SETH_F_LCLBYTEORDER == LDMS_SETH_F_LE
+	mv->v_d = v;
+#else
+	*(uint64_t *)&mv->v_d = __cpu_to_le64(*(uint64_t *)&v);
+#endif
+}
+
+void ldms_mval_array_set_str(ldms_mval_t mv, const char *str, size_t count)
+{
+	strncpy(mv->a_char, str, count);
+}
+
+void ldms_mval_array_set_char(ldms_mval_t mv, int idx, char v)
+{
+	mv->a_char[idx] = v;
+}
+
+void ldms_mval_array_set_u8(ldms_mval_t mv, int idx, uint8_t v)
+{
+	mv->a_u8[idx] = v;
+}
+
+void ldms_mval_array_set_s8(ldms_mval_t mv, int idx, int8_t v)
+{
+	mv->a_s8[idx] = v;
+}
+
+void ldms_mval_array_set_u16(ldms_mval_t mv, int idx, uint16_t v)
+{
+	mv->a_u16[idx] = __cpu_to_le16(v);
+}
+
+void ldms_mval_array_set_s16(ldms_mval_t mv, int idx, int16_t v)
+{
+	mv->a_s16[idx] = __cpu_to_le16(v);
+}
+
+void ldms_mval_array_set_u32(ldms_mval_t mv, int idx, uint32_t v)
+{
+	mv->a_u32[idx] = __cpu_to_le32(v);
+}
+
+void ldms_mval_array_set_s32(ldms_mval_t mv, int idx, int32_t v)
+{
+	mv->a_s32[idx] = __cpu_to_le32(v);
+}
+
+void ldms_mval_array_set_u64(ldms_mval_t mv, int idx, uint64_t v)
+{
+	mv->a_u64[idx] = __cpu_to_le64(v);
+}
+
+void ldms_mval_array_set_s64(ldms_mval_t mv, int idx, int64_t v)
+{
+	mv->a_s64[idx] = __cpu_to_le64(v);
+}
+
+void ldms_mval_array_set_float(ldms_mval_t mv, int idx, float v)
+{
+#if LDMS_SETH_F_LCLBYTEORDER == LDMS_SETH_F_LE
+	mv->a_f[idx] = v;
+#else
+	/* This type abuse is necessary to avoid the integer cast
+	 * that will strip the factional portion of the float value
+	 */
+	*(uint32_t *)&mv->a_f[idx] = __cpu_to_le32(*(uint32_t *)&v);
+#endif
+}
+
+void ldms_mval_array_set_double(ldms_mval_t mv, int idx, double v)
```

Comment:
@nichamon, same thing here only for doubles.

Reply:
Got it!